default/python/AI/ResourcesAI.py

```diff
@@ -114,7 +114,11 @@ def evaluate(self, focus: str) -> float:
         If focus produces best, return a positive rating, giving how much its output is better than second best.
         If another focus produces better, returns a negative rating, giving how much it is worse than best.
         """
+
         sqrt_population = min(1.0, self.planet.currentMeterValue(fo.meterType.population)) ** 0.5
+        if sqrt_population == 0:
+            sqrt_population = 0.0001  # prevent Zero division
```

Comment:
I understand this prevents zero division, but I really do not understand the rationale of using "min" here. For the majority of cases, it will mean that sqrt_population will equal 1, instead of square root of population, because for majority of cases the population will be higher than 1. I would rather see it as "max", which would mean if population is lower than 1, make it 1 for the purposes of this test. 

I did not follow the further logic of what is this doing in general, but it really screams bug at me if something called sqrt_population is sqrt_population only if the population was lower than 1 to begin with?

Reply:
> I understand this prevents zero division, but I really do not understand the rationale of using "min" here.

Same for me.  But somehow it worked before.  Maybe this code could be dropped without affecting AI behavior.


I want to fix a code error first since this breaks the AI behavior, and this is not good.
The second step is to fix the logic thing you mentioned.

Reply:
> > I understand this prevents zero division, but I really do not understand the rationale of using "min" here.
> 
> Same for me. But somehow it worked before. Maybe this code could be dropped without affecting AI behavior.
> 
> I want to fix a code error first since this breaks the AI behavior, and this is not good. The second step is to fix the logic thing you mentioned.

I suspect blueaward is right, and this was never right. I suspect "It worked before" = "It existed for 6 months without someone commenting". I'd fix it, but I can't tell if its right or wrong without walking through the logic step by step.